pkg/server/commands/write.go

```diff
@@ -71,12 +69,12 @@ func (c *WriteCommand) validateWriteRequest(ctx context.Context, req *openfgapb.
 			return err
 		}
 
-		typesys := typesystem.New(authModel)
-
-		if ProhibitModel1_0(typesys.GetSchemaVersion(), c.allowSchema10) {
-			return serverErrors.ValidationError(ErrObsoleteAuthorizationModel)
+		if !typesystem.IsSchemaVersionSupported(authModel.GetSchemaVersion()) {
```

Comment:
One part of me is thinking why don't we put this check inside typesystem.New?

Reply:
We can revisit this in a follow-up PR, but `typesystem.New` doesn't return an error. It assumes that the model is valid.

Reply:
`typesystem.NewAndValidate` takes the model, validates it, and returns either an error (if invalid) or the typesystem constructed from the model (if no error).

So yeah, I think having this as is right now makes the most sense. Those were by design.